browser/notifications/brave_notification_platform_bridge_mac.mm

```diff
@@ -0,0 +1,174 @@
+ * license. This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/notifications/brave_notification_platform_bridge.h"
+
+#import <Foundation/Foundation.h>
+#include <memory>
+
+#include "brave/browser/notifications/brave_alert_dispatcher_mac.h"
+#include "chrome/browser/notifications/notification_platform_bridge_mac.h"
+#include "ui/message_center/public/cpp/notification.h"
+#include "ui/message_center/public/cpp/notification_types.h"
+
+@class NSUserNotificationCenter;
+@class NSUserNotificationCenterDelegate;
+
+@interface NotificationTimeoutMac : NSObject
+
+- (void)startTimer:(NSUserNotification *)notification;
+
+@end
+
+@implementation NotificationTimeoutMac
+{
+    NSMutableArray<NSTimer *> * timers_;
+}
+
+- (id)init
+{
+    if (self = [super init]) {
+        timers_ = [[NSMutableArray alloc] init];
+    }
+
+    return self;
+}
+
+- (void)dealloc
+{
+    for (NSTimer * timer in timers_) {
+        [timer invalidate];
+        timer = nil;
+    }
+
+    [timers_ removeAllObjects];
+
+    [timers_ release];
+    timers_ = nil;
+
+    [super dealloc];
+}
+
+- (void)startTimer:(NSUserNotification *)notification
+{
+    NSTimer * timer = [NSTimer scheduledTimerWithTimeInterval:5.0
+                                                       target:self
+                                                     selector:@selector(timerFired:)
+                                                     userInfo:[notification retain]
+                                                      repeats:NO];
+    [timers_ addObject: timer];
+}
+
+- (void)timerFired:(NSTimer *)timer
+{
+    NSUserNotification* notification = timer.userInfo;
+    if (!notification) {
+        return;
+    }
+
+    auto* notification_center = [NSUserNotificationCenter defaultUserNotificationCenter];
+    [notification_center removeDeliveredNotification:notification];
+
+    [notification release];
+    notification = nil;
+
+    [timers_ removeObject: timer];
+}
+
+@end
+
+static NotificationTimeoutMac*
+    g_notification_platform_bridge_notification_timeout =
+        [[[NotificationTimeoutMac alloc] init] retain];
+
+@interface BraveNSUserNotificationCenter : NSUserNotificationCenter
+
+- (id)initWithProxy:(NSUserNotificationCenter *)proxy;
+- (void)setDelegate:(NSUserNotificationCenterDelegate *)delegate;
+- (void)deliverNotification:(NSUserNotification *)notification;
+- (void)removeDeliveredNotification:(NSUserNotification *)notification;
+- (void)removeAllDeliveredNotifications;
+- (NSArray<NSUserNotification *> *)deliveredNotifications;
+- (void)scheduleNotification:(NSUserNotification *)notification;
+- (NSArray<NSUserNotification *> *)scheduledNotifications;
+- (void)removeScheduledNotification:(NSUserNotification *)notification;
+
+@end
+
+@implementation BraveNSUserNotificationCenter
+{
```

Comment:
lint - opening brace should go at the end of the previous line for all of these

Reply:
Change to match Chromium style for objective c